parl/utils/logger.py

```diff
@@ -0,0 +1,148 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+import errno
+import os.path
+from termcolor import colored
+import sys
+
+__all__ = ['set_dir', 'get_dir', 'set_level']
+
+# globals: logger file and directory:
+LOG_DIR = None
+_FILE_HANDLER = None
+
+
+def _makedirs(dirname):
+    assert dirname is not None
+    if dirname == '' or os.path.isdir(dirname):
+        return
+    try:
+        os.makedirs(dirname)
+    except OSError as e:
+        if e.errno != errno.EEXIST:
+            raise e
+
+
+class _Formatter(logging.Formatter):
+    def format(self, record):
+        msg = '%(message)s'
+        if record.levelno == logging.WARNING:
+            date = colored(
+                '[%(asctime)s %(threadName)s @%(filename)s:%(lineno)d]',
+                'yellow')
+            fmt = date + ' ' + colored(
+                'WRN', 'yellow', attrs=['blink']) + ' ' + msg
+        elif record.levelno == logging.ERROR or record.levelno == logging.CRITICAL:
+            date = colored(
+                '[%(asctime)s %(threadName)s @%(filename)s:%(lineno)d]', 'red')
+            fmt = date + ' ' + colored(
+                'WRN', 'yellow', attrs=['blink']) + ' ' + msg
+            fmt = date + ' ' + colored(
+                'ERR', 'red', attrs=['blink', 'underline']) + ' ' + msg
+        elif record.levelno == logging.DEBUG:
+            date = colored(
+                '[%(asctime)s %(threadName)s @%(filename)s:%(lineno)d]',
+                'blue')
+            fmt = date + ' ' + colored(
+                'DEBUG', 'blue', attrs=['blink']) + ' ' + msg
+        else:
+            date = colored(
+                '[%(asctime)s %(threadName)s @%(filename)s:%(lineno)d]',
+                'green')
+            fmt = date + ' ' + msg
+        if hasattr(self, '_style'):
+            # Python3 compatibility
+            self._style._fmt = fmt
+        self._fmt = fmt
+        return super(_Formatter, self).format(record)
+
+
+def _getlogger():
+    logger = logging.getLogger('PARL')
+    logger.propagate = False
+    logger.setLevel(logging.DEBUG)
+    handler = logging.StreamHandler(sys.stdout)
+    handler.setFormatter(_Formatter(datefmt='%m-%d %H:%M:%S'))
+    logger.addHandler(handler)
+    return logger
+
+
+_logger = _getlogger()
+_LOGGING_METHOD = [
+    'info', 'warning', 'error', 'critical', 'warn', 'exception', 'debug',
+    'setLevel'
+]
+# export logger functions
+for func in _LOGGING_METHOD:
+    locals()[func] = getattr(_logger, func)
+    __all__.append(func)
+
+
+def _set_file(path):
+    global _FILE_HANDLER
+    if os.path.isfile(path):
+        os.remove(path)
+    hdl = logging.FileHandler(filename=path, encoding='utf-8', mode='w')
+    hdl.setFormatter(_Formatter(datefmt='%m-%d %H:%M:%S'))
+
+    _FILE_HANDLER = hdl
+    _logger.addHandler(hdl)
+
+
+def set_level(level):
```

Comment:
Passing a `string` as log level is ok, but it seems a bit curious because the API of logger is `_logger.setLevel(logging.WARNING)`
Could we use same argument type as that in _logger?